chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -707,8 +712,27 @@ impl EthereumAdapter {
             return Box::new(stream::empty());
         }
 
+        // Hit the database and fetch all failed transactions in the block range.
+        // TODO: handle this Result/expect
+        let mut transaction_statuses_in_block_range = {
+            stopwatch_metrics.start_section("calls_in_block_range__fetch_transaction_statuses");
+            chain_store
+                .transaction_statuses_in_block_range(&(from..to))
+                .expect("failed to fetch failed transactions in the database")
+        };
```

Comment:
You can do an early return with a `once` stream that emits an error https://docs.rs/futures/0.1.31/futures/stream/fn.once.html.

Reply:
Alternatively, we can log the error and move on as if we had nothing cached.